cypress-io/cypress
# feat: add `resourceType` support to `cy.intercept()` on `req`/`routeMatcher`

- Closes #14525

### User facing changelog

- Added the ability to match on `resourceType` with `cy.intercept()`, and to see the resource type of an intercepted request as `req.resourceType`.

### Additional details

- All browsers report this in pre-request information to enable proxy logging, so we can now safely add it in the public `intercept` API too.
- In combination with `cy.intercept(..., { log: false }`, these two changes will unblock users on disabling/enabling logs by resource type: https://github.com/cypress-io/cypress/issues/9358 (no PR yet for 9358)
- Also renamed "resourceType" to "requestedWith" everywhere it's used in multi-domain. Multi-domain was previously using the term "resource type" to refer to "xhr" or "fetch" for the runner patches, but this usage of the term clobbers the meaning given by CDP and makes it confusing. Since it's referred to as "requestedWith" in some places in MD already, I went ahead and renamed MD's usage to "requestedWith" for the entirety of MD.

### Steps to test

Use the new `resourceType` property on `RouteMatcher` and `req` to match only certain requests:

```ts
cy.intercept({ resourceType: 'stylesheet' }) // only CSS

cy.intercept('/my-api', (req) => {
  expect(req.resourceType).to.eq('xhr') // only XHRs should hit the API
})
```

Possible values:

```ts
type ResourceType = 'document' | 'fetch' | 'xhr' | 'websocket' | 'stylesheet' | 'script' | 'image' | 'font' | 'cspviolationreport' | 'ping' | 'manifest' | 'other'
```

### How has the user experience changed?


### PR Tasks


- [x] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? https://github.com/cypress-io/cypress-documentation/pull/4921
